tests/sentry/api/endpoints/test_project_rule_details.py

```diff
@@ -286,37 +207,57 @@ def test_with_jira_action_error(self):
             "frequency": 30,
         }
 
-        rule = Rule.objects.create(project=self.project, label="foo", data=data)
+        self.rule.update(data=data)
 
-        url = reverse(
-            "sentry-api-0-project-rule-details",
-            kwargs={
-                "organization_slug": self.project.organization.slug,
-                "project_slug": self.project.slug,
-                "rule_id": rule.id,
-            },
+        response = self.get_success_response(
+            self.organization.slug, self.project.slug, self.rule.id, status_code=200
         )
-
-        response = self.client.get(url, format="json")
-
-        assert response.status_code == 200
-
         # Expect that the choices get filtered to match the API: Array<string, string>
         assert response.data["actions"][0].get("dynamic_form_fields")[0].get("choices") == [
             ["EPIC-1", "Citizen Knope"],
             ["EPIC-2", "The Comeback Kid"],
         ]
 
 
-class UpdateProjectRuleTest(APITestCase):
-    @patch("sentry.signals.alert_rule_edited.send_robust")
-    def test_simple(self, send_robust):
-        self.login_as(user=self.user)
+class UpdateProjectRuleTest(ProjectRuleDetailsBaseTestCase):
+    method = "PUT"
 
-        project = self.create_project()
+    def assert_rule_from_payload(self, rule, payload):
```

Comment:
I'd pull this out of the class so that it's easier to share as a testhelper.

Reply:
Done, but I think I'll leave it in the same file since it was really only intended for the PUT method. It may need to be more abstract to be useful elsewhere.